tools/js-optimizer.js

```diff
@@ -7769,6 +7769,8 @@ function eliminateDeadGlobals(ast) {
 // it goes away.
 function JSDCE(ast) {
   var scopes = [{}]; // begin with empty toplevel scope
+  var isVarNameOrObjectKey;
```

Comment:
Actually, the first could be moved into the functions, no need to declare it this high?  the second needs to stay here, and could benefit from a comment